src/boards/rainbow_esp.cpp

```diff
@@ -365,35 +373,79 @@ void BrokeStudioFirmware::processBufferedMessage() {
 				this->writeFile(this->working_path, this->working_file, this->files[working_path][working_file].size(), this->rx_buffer.begin() + 3, this->rx_buffer.begin() + message_size + 1);
 			}
 			break;
-		case n2e_cmds_t::GET_FILE_LIST:
-			UDBG("RAINBOW BrokeStudioFirmware received message GET_FILE_LIST\n");
+		case n2e_cmds_t::FILE_COUNT:
+			UDBG("RAINBOW BrokeStudioFirmware received message FILE_COUNT\n");
 			if (message_size == 2) {
-				std::vector<uint8> existing_files;
 				uint8 const path = this->rx_buffer[2];
-				assert(this->file_exists[path].size() < 254);
-				for (uint8 i = 0; i < this->file_exists[path].size(); ++i) {
+				std::vector<uint8> existing_files;
+				this->tx_buffer.push_back(last_byte_read);
+				this->tx_buffer.push_back(2);
+				this->tx_buffer.push_back(static_cast<uint8>(e2n_cmds_t::FILE_COUNT));
+				if (path >= NUM_FILE_PATHS) {
+					this->tx_buffer.push_back(0);
+					break;
+				}
+				assert(this->file_exists[path].size() <= NUM_FILES);
```

Comment:
Strange assert. You are asserting that array size is lower or equal to the constant defining the array size. That should not be a runtime check as it is all compile time values. Actually if it is the intended check, I'd prefere to just not checking it.

Note, it is maybe intended as an update of the check `assert(this->file_exists[path].size() < 254);` which was certainly meant to ensure the following loop using an `uint8` will not overflow. if so, it is also the occasion to remove this overzealous check :)